### ISSUE TITLE
XML parse errors

### ISSUE THREAD

**Issue by stalakerob** [mezz64/pyHik#40]
I have a Hikvision DS-2CD2045FWD-I cam with firmware V5.6.0 build 190507. I'm getting XML parse errors. The cam is hardwired using a powerline connection. The connection is pretty stable providing ~ 75Mbit/s. I'm also using Hikvision's iVMS-4500 app on my Android smartphone. That way I can compare results. During testing I'm receiving all motion triggers on the phone and also pyHik receives them but generates lots of XML parse errors.

**Comment by stalakerob** [comment]
How does pyHik handle this case currently? Are both events discarded? An alternative could be that pyHik restarts event processing when the interrupting event arrives and discards the interrupted event; 
Does that make sense?

**Comment by mezz64** [comment]
At the moment both events are discarded since the event string is only cleared after an ending event is found.  A simple fix would be to clear the event string when a start event is found. 

You can test this by adding the below after line 485:
https://github.com/mezz64/pyHik/blob/dfc0cee72ec1eb2c73a6dd2598e8727ed137179a/pyhik/hikvision.py#L485

```
parse_string = ""
```

**Comment by stalakerob** [comment]
In the meantime I observed the following: I had expert mode turned on for motion detection and defined multiple (4) areas with different sensitivity. That leads to a pretty long data stream when it triggers which is sometimes interrupted by another event as we know. When turning expert mode off, the data stream for that event is much shorter. I've tested with expert mode off for a while and did not see any jumbled messages any more and therefore no XML parse errors either. So that could be the solution for those who do not need expert mode.
For those who need expert mode your proposed code change above might be a workaround. I'll have a look at it later today or tomorrow and let you know if it works. 
Thanks, Robert

**Comment by stalakerob** [comment]
Clarifying question: you mean changing the 2 lines starting at #485:

`start_event = True`
`parse_string += str_line` 

to the following 3 lines?

`start_event = True`
`parse_string = ""`
`parse_string += str_line`

**Comment by madprunes** [comment]
I don't think this has to do with events interrupting each other, you will notice it only occurs on the really long event messages (typically around 2037 characters), looks like there is a cutoff where any lines which don't fit within the message limit (probably 2048 characters) just get dropped.

It looks like enough data is there you could attempt to parse the unfinished event data but it's far from ideal, the best way would be figuring how to reduce the <RegionCoordinates> tags from the camera, if that is even possible?

**Comment by mezz64** [comment]
I added a change to the recently release 0.2.4 version that will at least catch the last event when they get jumbled together.  I'll leave this issue open for now since that's only a partial fix.
